src/trainer.py

```diff
@@ -312,7 +317,11 @@ def train(self, tasks, stop_metric,
         task_infos, metric_infos = self._setup_training(tasks, batch_size, train_params,
                                                         optimizer_params, scheduler_params, phase)
 
-        if shared_optimizer:  # if shared_optimizer, ignore task_specific optimizers
+        if shared_optimizer:  # If shared_optimizer, ignore task_specific optimizers
+            if self._max_epochs_per_task:
+                # TODO(Alex or Yada): make this generalizable to not depend
```

Comment:
what does this mean?

Reply:
Actually revert this, I think this was from when I was trying to figure out how _max_epochs_per_task worked